src/sfma/models/marginal.py

```diff
@@ -90,20 +100,18 @@ def objective(self, x: ndarray, data: Data) -> float:
         v = data.obs_var + v_re + v_ie
         z = np.sqrt(v_ie)*r/np.sqrt(2.0*v*(data.obs_var + v_re))
 
-        return np.mean(0.5*r**2/v + 0.5*np.log(v) - log_erfc(z))
+        return 0.5 * r ** 2 / v + 0.5 * np.log(v) - log_erfc(z)
 
-    def gradient(self, x: ndarray, data: Data) -> ndarray:
-        """
-        Computes the gradient.
+    def objective(self, x: ndarray, data: Data) -> float:
+        obj = self._objective(x=x, data=data)
+        if self.w is not None:
+            obj = self.w.dot(obj)
```

Comment:
!Inconsistency with the gradient below, here we didn't take the mean. When `self.w is not None`, `self.w.dot(obj)` will be a scalar, the mean below won't do anything. Imagine the case when `self.w` are a vector with all ones.